### PR TITLE
Fix/image order

### PR DESCRIPTION
**Pull request by knzwats**
### Summary

If there is a sheet without any images attached, indices for image files are messed up and crammed.

 I had to have a proper sheet indices as a hash and later refer to it to correctly resolve the sheet that the image belongs.

My code might look wonky but it works!

here's the rake test result:
```
Finished in 9.867465s, 16.9243 runs/s, 222.4482 assertions/s.

167 runs, 2195 assertions, 0 failures, 0 errors, 7 skips

You have skipped tests. Run with --verbose for details.
Coverage report generated for Unit Tests to /Users/atsushikanazawa/Projects/roo/coverage. 1656 / 1898 LOC (87.25%) covered.

```

### Other Information

I attached an excel file that contains 2 kangaroos on sheet 1 and 3.
If you run it with older code, it tells you that sheet 1 and 2 have kangaroos.

Thanks for contributing to Roo!

### CODE REVIEW

File: test/image_order/test_image_order.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require "test_helper"
+require 'pry'
```
Comment (chopraanmol1) [inline test/image_order/test_image_order.rb]: Remove this line

**Comment by knzwats** [comment]
will do, thanks!

**Comment by knzwats** [comment]
I fixed the code!

But I want to elaborate test cases a little bit more when I get to office(I don't have Excel installed on any of my cumputers thus cannot modify excel files at home). 

Any test scenarios do you have in your mind?
Thanks!

File: test/image_order/test_image_order.rb (hunk 1 above)
Comment (knzwats) [reply test/image_order/test_image_order.rb]: done

File: lib/roo/excelx/relationships.rb [hunk 2]
```diff
@@ -19,6 +19,12 @@ def include_type?(type)
         end
       end
 
+      def target(type)
+        to_a.map do |_, rel|
+          rel["Target"] if rel["Type"]&.include? type
+        end.compact.first
+      end
```
Comment (chopraanmol1) [inline lib/roo/excelx/relationships.rb]: How about implementing something like following (with better method name):
```
def targets(type: "")
  return to_enum(:targets, type: type) unless block_given?

  to_a.each do |_, rel|
    yield rel["Target"] if rel["Type"]&.include? type
  end
end
```

File: lib/roo/excelx/relationships.rb [hunk 3]
```diff
@@ -19,6 +19,12 @@ def include_type?(type)
         end
       end
 
+      def target(type)
```
Comment (chopraanmol1) [inline lib/roo/excelx/relationships.rb]: `target` is a confusing name for what this method is doing.

We also need relevant test cases for the newly added method.

File: lib/roo/excelx/sheet.rb [hunk 4]
```diff
@@ -96,6 +93,16 @@ def dimensions
         @sheet.dimensions
       end
 
+      def images
```
Comment (chopraanmol1) [inline lib/roo/excelx/sheet.rb]: This method should be memoized.
```
def images
  @images ||= begin
     ...
  end
end
```

File: lib/roo/excelx/sheet.rb [hunk 5]
```diff
@@ -96,6 +93,16 @@ def dimensions
...
+        target = @rels.target("drawing")
+        match = /[a-zA-Z]+([0-9]+).xml/.match target
+        if match
```
Comment (chopraanmol1) [inline lib/roo/excelx/sheet.rb]: Do we need this condition? Images class already handles nil as file_path

File: lib/roo/excelx/relationships.rb (hunk 3 above)
Comment (chopraanmol1) [reply lib/roo/excelx/relationships.rb]: The method name should signify that we are returning **target(s)** while filtering relations on **Type** attribute. If it doesn't signify the above then it is confusing in nature.

File: lib/roo/excelx.rb [hunk 6]
```diff
@@ -455,7 +455,6 @@ def process_zipfile_entries(entries)
           nr = Regexp.last_match[1].to_i
           image_rels[nr - 1] = "#{@tmpdir}/roo_image_rels#{nr}"
         end
-
```
Comment (chopraanmol1) [inline lib/roo/excelx.rb]: Revert unnecessary cosmetic changes

Comment (knzwats) [reply lib/roo/excelx.rb]: sorry about that

**Comment by knzwats** [comment]
Fixed!

But I have few questions.

I just noticed the order of image resources are like this if you have multiple images:

```
{"rId3"=>"roo_media_image3.jpeg", "rId2"=>"roo_media_image2.jpeg", "rId1"=>"roo_media_image1.jpeg", "rId4"=>"roo_media_image4.jpeg"}
```

It looks like it's random and I thought we'd better sort it by its number, so I coded just so it does.
Sorting a hash does not make much sense, but `sheet#images` method is only used inside `excelx#images` as far as I grepped, and keys are discarded so it is effectively an array, because keys are not exposed anyway:

```
def images(sheet = nil)
  images_names = sheet_for(sheet).images.map(&:last)
  images_names.map { |iname| image_files.find { |ifile| ifile[iname] } }
end
```

I bumped into this problem in my job and sorted images based on its names outside of roo.
Do you think sorting is necessary? 

And I'd love to write some tests on `relationship#targets` method.
Should I add new test method to `roo/excelx/relationships_spec.rb` or create a brand new testcase?
Is there any excelx file or specific scenario to test `relationship#targets` in your mind because I do not know roo's internal design or its edge cases.

Thanks!
@chopraanmol1

File: lib/roo/excelx/relationships.rb (hunk 2 above)
Comment (knzwats) [reply lib/roo/excelx/relationships.rb]: I was thinking there is always a single target for a type.
changed name and implemented according to your proposla.

**Comment by knzwats** [comment]
hello?

File: lib/roo/excelx/sheet.rb [hunk 7]
```diff
@@ -96,6 +93,15 @@ def dimensions
         @sheet.dimensions
       end
 
+      def images
+        @images ||= begin
+          @rels.filter_by_type(type: "drawing").map do |target|
+            match = /[a-zA-Z]+([0-9]+).xml/.match target
+            Images.new(image_rels[match[1].to_i - 1]).list
+          end.reduce({}, :merge).sort.to_h
```
Comment (simonoff) [inline lib/roo/excelx/sheet.rb:101]: What if it fails?
